phpunit/class-wp-theme-json-test.php

```diff
@@ -751,6 +751,30 @@ public function test_get_stylesheet_with_block_support_feature_level_selectors()
 		$this->assertEquals( $expected, $theme_json->get_stylesheet() );
 	}
 
+	public function test_allow_inferred_properties() {
+		$actual = WP_Theme_JSON_Gutenberg::remove_insecure_properties(
+			array(
+				'version' => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
+				'styles'  => array(
+					'spacing' => array(
+						'blockGap' => '3em',
+					),
+				),
+			)
+		);
+
+		$expected = array(
+			'version' => WP_Theme_JSON_Gutenberg::LATEST_SCHEMA,
+			'styles'  => array(
+				'spacing' => array(
+					'blockGap' => '3em',
+				),
+			),
+		);
+
+		$this->assertEqualSetsWithIndex( $expected, $actual );
```

Comment:
Would we need to cover block-level spacing as well or is that being piped through the same mechanism?

Reply:
Turns out I think I made a typo when I tried the block-level spacing in this test earlier, but it's working now that I've written it out correctly 😄. Added in https://github.com/WordPress/gutenberg/pull/46388/commits/5b629c87a04d2c18171b690348d7515e31b04119.